pkg/profile/git_repo.go

```diff
@@ -0,0 +1,45 @@
+package profile
+
+import (
+	"strings"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GitRepoRequiresUpdate checks if the git repository resource requires updating
+func GitRepoRequiresUpdate(existingRes, newRes *sourcev1.GitRepository) bool {
+	switch {
+	case existingRes.Spec.URL != newRes.Spec.URL:
+		return true
+	case existingRes.Spec.Reference.Branch != newRes.Spec.Reference.Branch:
+		return true
+	default:
+		return false
+	}
```

Comment:
```suggestion
	return existingRes.Spec.URL != newRes.Spec.URL || existingRes.Spec.Reference.Branch != newRes.Spec.Reference.Branch
```

Reply:
I think this makes it harder to read. The select is more lines, but much more readable